### PR TITLE
Fix Bug in Social Authentication Logic

### PR DESCRIPTION
**Pull request by elemanhillary-zz**
#### What does this PR do?
- fix redirects to the frontend
#### Description of Task to be completed
- enables redirects to the frontend

#### How should this be manually tested
```
git clone git@github.com:andela/ah-92explorers-backend.git
cd ah-92explorers-backend
git checkout bg-fix-redirect-168078314
```
2. run `npm install`

3. Install your local database in Postgres

4. Rename `.env.sample` to `.env` and `CRYPTO_PASSSWORD` enviroment variable

5. run `npm run start`

6. Using Google Chrome (or any other browser) 
- `http://localhost:3000/api/auth/facebook`
- `http://localhost:3000/api/auth/google`
- `http://localhost:3000/api/auth/twitter`

#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#168078314](https://www.pivotaltracker.com/story/show/168078314)
#### Screenshots (if appropriate)
- None
#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

File: controllers/socialAuth.js [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
 
 dotenv.config();
-
+const { FRONT_END_URL } = process.env;
 const { users } = models;
-
+const algorithm = 'aes-256-ctr';
+const { CRYPTO_PASSWORD } = process.env;
 export default class User {
+  static enc(data) {
```
Comment (IsaiahRn) [inline controllers/socialAuth.js:12]: What were you trying to achieve here?

**Review by niyoceles — APPROVED** [review]

**Review by niyoceles — DISMISSED** [review]
@elemanhillary remember to talk with @Hervera  to add this variable `FRONT_END_URL` on Heroku, Share to your member the value of this variable CRYPTO_PASSWORD

File: controllers/socialAuth.js (hunk 1 above)
Comment (elemanhillary-zz) [reply controllers/socialAuth.js:12]: atleast securing the return object by encrypting it and sending it to the frontend as a token

Comment (IsaiahRn) [reply controllers/socialAuth.js:12]: Okay!

**Review by IsaiahRn — DISMISSED** [review]

**Review by niyoceles — APPROVED** [review]
The bug has been fixed

**Review by kodek-sleuth — APPROVED** [review]
This PR is ready to be merged

**Review by IsaiahRn — APPROVED** [review]
This PR is good to be merged

**Pull request merged**